test/tools/ossfuzz/YulToEvmDifferentialFuzzer.cpp

```diff
@@ -0,0 +1,251 @@
+			callResultOpt.status_code != EVMC_REVERT,
+			"SolidityEvmoneInterface: EVM One reverted"
+			);
+	if (noInvalidInSource)
+		solAssert(
+			callResultOpt.status_code != EVMC_INVALID_INSTRUCTION,
+			"Invalid instruction."
+			);
+	solAssert(
+		(callResultOpt.status_code == EVMC_SUCCESS ||
+		(!noRevertInSource && callResultOpt.status_code == EVMC_REVERT) ||
+		(!noInvalidInSource && callResultOpt.status_code == EVMC_INVALID_INSTRUCTION)),
+		"Call failed."
+		);
+	_state << EVMHostPrinter{_host, deployResultOpt.create_address}.storageOnly();
+}
+}
+
+static evmc::VM evmone = evmc::VM{evmc_create_evmone()};
+
+DEFINE_PROTO_FUZZER(Program const& _input)
+{
+	// Solidity creates an invalid instruction for subobjects, so we simply
+	// ignore them in this fuzzer.
+	if (_input.has_obj())
+		return;
+	ProtoConverter converter;
+	string yulSubObject = converter.programToString(_input);
+	langutil::EVMVersion version;
+	EVMHost hostContext(version, evmone);
+
+	YulStringRepository::reset();
+
+	// Package test case into a sub-object
+	Whiskers yulObjectFormat(R"(
+	object "main" {
+		code {
+			codecopy(0, dataoffset("deployed"), datasize("deployed"))
+			return(0, datasize("deployed"))
+		}
+		object "deployed" {
+			code {
+				<fuzzerInput>
+			}
+		}
+	}
+		)");
+	string yul_source = yulObjectFormat("fuzzerInput", yulSubObject).render();
+
+	if (const char* dump_path = getenv("PROTO_FUZZER_DUMP_PATH"))
+	{
+		ofstream of(dump_path);
+		of.write(yul_source.data(), static_cast<streamsize>(yul_source.size()));
+	}
+
+	solidity::frontend::OptimiserSettings settings = solidity::frontend::OptimiserSettings::none();
+	AssemblyStack yulStack(version, AssemblyStack::Language::StrictAssembly, settings);
+	solAssert(
+		yulStack.parseAndAnalyze("source", yulSubObject),
+		"Parsing fuzzer generated input failed."
+	);
+	std::shared_ptr<yul::Object> subObject = yulStack.parserResult();
+	Dialect const& dialect = EVMDialect::strictAssemblyForEVMObjects(version);
+	string unoptimizedObject = Whiskers(R"(
+	object "main" {
+		code {
+			codecopy(0, dataoffset("deployed"), datasize("deployed"))
+			return(0, datasize("deployed"))
+		}
+		object "deployed" {
+			code {
+				<fuzzerInput>
+			}
+		}
+	}
+	)")
+		("fuzzerInput", yulSubObject)
+		.render();
+
+	bytes unoptimisedByteCode;
+	try
+	{
+		unoptimisedByteCode = YulAssembler{version, settings, unoptimizedObject}.assemble();
+	}
+	catch (yul::StackTooDeepError const&)
+	{
+		return;
+	}
+
+	ostringstream unoptimizedState;
+	runOnce(hostContext, unoptimisedByteCode, unoptimizedState, yulSubObject);
+
+	YulOptimizerTestCommon optimizerTest(
+		subObject,
+		dialect
+	);
+	// Run circular references pruner and then stack limit evader.
+	string step = "fullSuite";
+	optimizerTest.setStep(step);
+	shared_ptr<solidity::yul::Block> astBlock = optimizerTest.run();
+	bool recursiveFunction = recursiveFunctionHasUnreachableVariables(dialect, *subObject);
+	string optimisedSubObject = AsmPrinter{}(*astBlock);
```

Comment:
Ah, ok, so you do run the full Yul optimizer for the second run, but not the first... I'll need to check, if given that, I can reprodue the stack error. But not sure we should do that in general, at least for a first comparison.
For a first comparison it might be good to just run the code transform with and without stack optimization on the very same yul code.

Then we can check, if we want to force the old stack optimizations of the old code transform as reference (that will allow comparing more cases in which the old transform will otherwise throw stack too deep), that would involve replacing ``YulAssembler{...}.assemble();`` by something more manual for the first case.

Then we can check what happens with the optimizer enabled in both cases.

Reply:
> For a first comparison it might be good to just run the code transform with and without stack optimization on the very same yul code.

Okay, I'll refactor the Yul (full suite) optimized into an unused local function for now (essentially, comment out the said lines) and rerun the fuzzer. Does that sound reasonable?

Reply:
Yep

Reply:
Still, what worries me a bit, is that for the test case you posted earlier, I don't seem to be able to get a stack-too-deep error no matter what combination I try :-).

Reply:
Also I'd still be interested in any new exceptions thrown that are not stack-too-deep errors.

Reply:
Can I run stack optimizer on Yul objects of the format 

```
object "main" {
  code {
    <deployObject>
  }
}
object "deployedObject" {
  code {
    <FuzzerGeneratedCode>
  }
}
}
```

Reply:
Should be fine, yeah.